packages/shared/src/permissions/permissionsClass.ts

```diff
@@ -649,6 +651,62 @@ export class Permissions {
     );
   };
 
+  //TODO: Refactor this into two separate methods and eliminate updating envs from organizations.controller.putOrganization - Github Issue #2494
+  public canCreateOrUpdateEnvironment = (
+    environment: Pick<Environment, "projects" | "id">
+  ): boolean => {
```

Comment:
Is this properly handling the case when the list of projects changes during an update?  If that's being handled upstream, that's fine for now.  Eventually, we'll want to move that logic here.

Reply:
It does - before we're showing the edit button, we check if the user can update the existing env, and then when we make the call to update, we're only passing in the updated `env`, and so if the projects change, it'll get caught here.